xarray/coding/cftime_offsets.py

```diff
@@ -0,0 +1,765 @@
+"""Time offset classes for use with cftime.datetime objects"""
+import re
+
+from datetime import timedelta
+from functools import partial
+
+import numpy as np
+import pandas as pd
+
+from .cftimeindex import _parse_iso8601_with_reso, CFTimeIndex
+from .times import format_cftime_datetime
+from ..core.pycompat import basestring
+
+try:
+    from pandas.errors import OutOfBoundsDatetime
+except ImportError:
+    # pandas < 0.20
+    from pandas.tslib import OutOfBoundsDatetime
+
+
+def get_date_type(calendar):
+    """Return the cftime date type for a given calendar name."""
+    try:
+        import cftime
+    except ImportError:
+        raise ImportError(
+            'cftime is required for dates with non-standard calendars')
+    else:
+        calendars = {
+            'noleap': cftime.DatetimeNoLeap,
+            '360_day': cftime.Datetime360Day,
+            '365_day': cftime.DatetimeNoLeap,
+            '366_day': cftime.DatetimeAllLeap,
+            'gregorian': cftime.DatetimeGregorian,
+            'proleptic_gregorian': cftime.DatetimeProlepticGregorian,
+            'julian': cftime.DatetimeJulian,
+            'all_leap': cftime.DatetimeAllLeap,
+            'standard': cftime.DatetimeProlepticGregorian
+        }
+        return calendars[calendar]
+
+
+class BaseCFTimeOffset(object):
+    _freq = None
+
+    def __init__(self, n=1):
+        self.n = n
```

Comment:
do you want to add checks here to verify that `n` is an integer? or maybe floats are OK, too?

Reply:
Indeed `n` must be an integer.  I added a check here.  I also added a check on the `month` argument provided to the `YearOffset` classes.